File: server/src/models/Event.js
/**
 * Pseudocode:
 * 1) Document Event model shape.
 * 2) Persist telemetry events to storage.
 */

/**
 * @typedef {Object} Event
 * @property {string} eventId
 * @property {string} sessionId
 * @property {string} timestamp
 * @property {number} stageId
 * @property {string} eventType
 * @property {Object<string, unknown>} payload
 * @property {string} configId
 * @property {boolean=} isValid
 */

export const EVENT_MODEL_DOCS = {};

File: server/src/models/Event.java
package models;

import java.util.Map;

/**
 * Pseudocode:
 * 1) Document Event model shape.
 * 2) Persist telemetry events to storage.
 */
public class Event {
  public String eventId;
  public String sessionId;
  public String timestamp;
  public Integer stageId;
  public String eventType;
  public Map<String, Object> payload;
  public String configId;
  public Boolean isValid;
}
